daemon/cmd/agenthealth.go

```diff
@@ -65,13 +70,20 @@ func (d *Daemon) startAgentHealthHTTPService(addr string) {
 			}
 			return false
 		}
+		currentTs := d.svc.GetCurrentTs()
+		var lastUpdatedTs time.Time
+		lastUpdatedTs = currentTs
 
 		statusCode := http.StatusOK
 		sr := d.getStatus(true)
 		if isUnhealthy(&sr) {
+			// If unhealthy, return the timestamp of the last update.
+			lastUpdatedTs = d.svc.GetLastUpdatedTs()
 			statusCode = http.StatusInternalServerError
 		}
+
 		w.WriteHeader(statusCode)
+		fmt.Fprintf(w, `{"lastUpdated": %q, "currentTime": %q}`, lastUpdatedTs, currentTs)
```

Comment:
Are breaking here some users who rely on Cilium/healthz? Is the function comment still accurate re this being equivalent of 'cilium status --brief'?

Reply:
This comment was based on what kube-proxy states. Not sure what you meant.

Reply:
I think I got the point, what you mentioned, I can use the 'addr' to distinguish the request for and based on that only provide the 'Upatedtime and currenttime' for the kube-proxy listener address and not for cilium.